csrc/transform_rfactor.cpp

```diff
@@ -241,7 +228,7 @@ class ReplayRFactor : public ReplayTransformations {
   // The updated domain matching the producer's logical domain. This rfactor
   // domain is relative to the iter domains in the origianl_domain and must be
   // updated to grab the mapped id's later.
-  std::vector<IterDomain*> logical_domain_;
+  LinkedHashMap<IterDomain*, std::monostate> logical_domain_;
```

Comment:
Am I understanding correctly that `std:monostate` is used here because we don't care the mapping part? We just want to have something like `LinkedHashSet`?

Reply:
Yes. I can create a separate data structure too

Reply:
You might be right. We were doing 3 finds before and one now, but the benefit may not be huge.

Reply:
I suspect it could be actually slower. `std::find` should be extremely cheap operation since the comparison is just pointer equivalence. With `std::vector`, it wouldn't need to maintain the hash map and the linked list.

Reply:
I felt find is less of a problem.

I can see std::vector with insert/erase could cause some data movement, but as Naoya pointed out, for a smaller data structure, it's probably fine.

I felt std::list could be a better choice. 😉 But unless we have data showing that this is a problem, I'm happy with LinkedHashMap as-is.